### PR TITLE
Require webpack 5 in our lib. Add minimum webpack and typescript versions to storybook. Update `html-to-parser` version

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
* Update `html-to-parser` version to the latest version that supports react 18
* Add documentation for minimum typescript and webpack support
* Update storybook manager to use webpack 5 (to avoid https://github.com/fb55/htmlparser2/issues/1237)
* Update tools/check-treeshaking to use babel workaround for: https://github.com/fb55/htmlparser2/issues/1237

# Why
Continuation of #2415 and #2427 to support react 18.

# How Tested
Verified storybook ran locally.

### CODE REVIEW

File: packages/storybook/stories/Troubleshooting.stories.mdx
Comment (prprabhu-ms) [inline packages/storybook/stories/Troubleshooting.stories.mdx:82]:
```suggestion
If this is not possible, take a look at [these workarounds](https://github.com/fb55/htmlparser2/issues/1237#issuecomment-1182522861).
```

**Review by prprabhu-ms — APPROVED** [review]

**Comment by prprabhu-ms** [comment]
How do you feel about holding this PR back until we cut the next stable release branch. It's been a while since the last stable release, and it'd be good to release this in a separate, smaller, stable release, clearly calling out the change.

**Comment by JamesBurnside** [comment]
> How do you feel about holding this PR back until we cut the next stable release branch. It's been a while since the last stable release, and it'd be good to release this in a separate, smaller, stable release, clearly calling out the change.

Sure thing!

**Comment by prprabhu-ms** [comment]
Feel free to merge this when ready. `1.4.0` release branch has been created: https://github.com/Azure/communication-ui-library/tree/release/1.4.0

File: packages/storybook/package.json
Comment (PorterNan) [inline packages/storybook/package.json:56]: There are some other packages like @storybook/api are still staying in ~6.5.7, I guess we might want to align them together? (maybe make them all into ~6.5.12?)

**Review by PorterNan — APPROVED** [review]

File: packages/storybook/package.json
Comment (JamesBurnside) [reply packages/storybook/package.json:56]: We probably should but as there's no errors I imagine it will fall into 'isn't broke won't fix it'. In saying that storybook7 is currently in development and on like alpha v42 so easiest to wait for that then do a big upgrade!

**Pull request merged**
